pkg/project/operations.go

```diff
@@ -190,37 +191,45 @@ func List(ctx context.Context, opts Options) (projects []string, warnings map[st
 	return projects, warnings, nil
 }
 
-func GetDetails(ctx context.Context, opts Options, projectNames []string) (result []DetailProject, err error) {
-	entryCh := make(chan DetailProject)
-	errCh := make(chan error)
+func GetDetails(ctx context.Context, opts Options, projectNames []string) (projects []DetailProject, err error) {
+	var (
+		wg     sync.WaitGroup
+		result = make(chan DetailProject)
+	)
+
+	getProjectDetails(ctx, &wg, result, opts, projectNames)
+
+	go func() {
+		wg.Wait()
+		close(result)
+	}()
+
+	for detailProject := range result {
+		projects = append(projects, detailProject)
+	}
 
-	// Launch a goroutine for each project to retrieve its information
+	return projects, nil
+}
+
+func getProjectDetails(ctx context.Context, wg *sync.WaitGroup, result chan<- DetailProject, opts Options, projectNames []string) {
 	for _, projectName := range projectNames {
-		go func(proj string, opts Options) {
-			opts.Project = proj
+		wg.Add(1)
+		go func(projectName string, opts Options) {
+			defer wg.Done()
+			// Launch a goroutine for each project to retrieve its information
+			opts.Project = projectName
 			c, err := Client(ctx, opts)
 			if err != nil {
-				errCh <- err
+				logrus.Debugf("unable to get client for %s: %s", projectName, err)
```

Comment:
nit: should this be a warn level log? or possibly let this bubble up in a `DetailProject.Err` with a `nil` `Project` field?

Reply:
yeah I was trying to follow the pattern of the List func where it just debug logs when local kubeconfig client is invalid but I think it makes more sense to warn here instead in this case.